compiler/ifa/num.h

```diff
@@ -110,20 +118,24 @@ class Immediate { public:
   }
   Immediate& operator=(char *s) {
     const_kind = CONST_KIND_STRING;
+    string_kind = STRING_KIND_C_STRING;
     v_string = s;
     return *this;
   }
-  Immediate(bool b) {
-    memset(this, 0, sizeof(*this));
-    const_kind = NUM_KIND_BOOL;
-    num_index = BOOL_SIZE_SYS;
+  Immediate(bool b) :
+    const_kind(NUM_KIND_BOOL),
+    num_index(BOOL_SIZE_SYS)
```

Comment:
Why did the memset go away? Deemed unnecessary? It looks OK, but I'd leave it in personally (adding fields is easier since they will be zero'd)